### Упражнения

__Задача 1__

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите `M(X)`, `D(X)`.

вероятность угадать ответ P(A) = 0.25

мы можем угадать 0, 1, 2, 3, 4 или 5 ответов

используем биноминальный закон распределения

$$P(X = 0) = C_n^k p^k (1 - p)^{n - k} = C_5^0 \cdot 0.25^0 \cdot 0.75^5 = \frac{5!}{0! \: (5-0)!} \cdot 0.25^0 \cdot 0.75^5 = 1 \cdot 1 \cdot 0.237 = 0.237.$$

$$P(X = 1) = C_n^k p^k (1 - p)^{n - k} = C_5^1 \cdot 0.25^1 \cdot 0.75^4 = \frac{5!}{1! \: (5-1)!} \cdot 0.25^1 \cdot 0.75^4 = 5 \cdot 0.25 \cdot 0.316 = 0.395.$$

$$P(X = 2) = C_n^k p^k (1 - p)^{n - k} = C_5^2 \cdot 0.25^2 \cdot 0.75^3 = \frac{5!}{2! \: (5-2)!} \cdot 0.25^2 \cdot 0.75^3 = 0.264.$$

$$P(X = 3) = C_n^k p^k (1 - p)^{n - k} = C_5^3 \cdot 0.25^3 \cdot 0.75^2 = \frac{5!}{3! \: (5-3)!} \cdot 0.25^3 \cdot 0.75^2 = 0.088.$$

$$P(X = 4) = C_n^k p^k (1 - p)^{n - k} = C_5^4 \cdot 0.25^4 \cdot 0.75^1 = \frac{5!}{4! \: (5-4)!} \cdot 0.25^4 \cdot 0.75^1 = 0.015.$$

$$P(X = 5) = C_n^k p^k (1 - p)^{n - k} = C_5^5 \cdot 0.25^5 \cdot 0.75^0 = \frac{5!}{5! \: (5-5)!} \cdot 0.25^5 \cdot 0.75^0 = 0.001.$$

закон распределения такой случайной величины:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = x)$</th>
        <td>0.237</td>
        <td>0.395</td>
        <td>0.264</td>
        <td>0.088</td>
        <td>0.015</td>
        <td>0.001</td>
    </tr>
</table>

Посчитаем математическое ожидание распределения. Поскольку $n = 5$ и $p = 0.25$, получаем:

$$M(X) = 5 \cdot 0.25 = 1.25.$$

Дисперсия такой величины:

$$D(X) = 5 \cdot 0.25 \cdot (1 - 0.25) = 0.94.$$

__Задача 2__

Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

 рассматриваемая случайная величина имеет распределение Пуассона, т.к. мы можем посчитать число попаданий. В нашем случае $a = 200 \cdot 0.01 = 2$
 
 $$P(X = 5) = \dfrac{2^{5} \cdot e^{-2}}{5!} \approx 0.0361.$$
 
 $$P(X = 6) = \dfrac{2^{6} \cdot e^{-2}}{6!} \approx 0.0120.$$
 
 $$P(X = 7) = \dfrac{2^{7} \cdot e^{-2}}{7!} \approx 0.0034.$$
 
 $$P(X = 8) = \dfrac{2^{8} \cdot e^{-2}}{8!} \approx 0.0009.$$
 
 $$P(X = 9) = \dfrac{2^{9} \cdot e^{-2}}{9!} \approx 0.0002.$$
 
 $$P(X = 10) = \dfrac{2^{10} \cdot e^{-2}}{10!} \approx 0.00004.$$
 
 находим сумму вероятностей? p = 0.05264

In [25]:
import numpy as np

a = 2
k = 10

(a ** k) * (np.exp(-a)) / np.math.factorial(k)

3.8189850648779595e-05

In [26]:
0.0361 + 0.0120 + 0.0034 + 0.0009 + 0.0002 + 0.00004

0.05264

__Задача 3__

В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?

_Подсказка_. Как посчитать `P(X >= k)` для величины `X`, имеющей распределение Пуассона? Здесь предстоит немножко покодить.

число выстрелов при заданной вероятности влияет на параметр а в распределении Пуассона. найдем n при котором вероятность P(X >= k) будет = 0,9

In [42]:
n = 1
k = 10
f = np.math.factorial(k)

old_p = -1
while True:
    a = n * 0.01
    p = (a ** k) * (np.exp(-a)) / f
    if p + 0.0001 > 0.1:
        break
    n += 1

print(p, n)

0.10000177996636454 803


за 803 выстрела мы попадем 10 раз с вероятностью 0,1. как это докрутить до 0,9 не совсем понятно.